src/sdltiles.cpp

```diff
@@ -3781,6 +3781,15 @@ window_dimensions get_window_dimensions( const point &pos, const point &size )
     return get_window_dimensions( {}, pos, size );
 }
 
+bool window_contains_point( const catacurses::window &win, const point p )
+{
+    const window_dimensions dim = get_window_dimensions( win );
+    const point &win_min = dim.window_pos_pixel;
+    const point &win_size = dim.window_size_pixel;
```

Comment:
This function compares `p` with the pixel position of the window, but is used at inventory_ui.cpp:1889 to compare text position.

Reply:
sorry that was a mistake

Reply:
You can compare `p` with `dim.window_pos_cell` and `dim.window_size_cell` instead, or you can compare `p` with `getbegx(win)`, `getbegy(win)`, `getmaxx(win)`, and `getmaxy(win)`.

Reply:
I still cannot understand why here need to use half_open_rectangle, isn't it supposed to be inclusive_rectangle?

Reply:
`half_open_retangle` treats the maximum value outside the rectangle, while `inclusive_rectangle` treats the maximum value inside the rectangle. So if the maximum value is `pos + size`, then you use `half_open_rectangle` since `pos + size` is outside of the rectangle (the actual maximum position is `pos + size - (1, 1)`. If the maximum value is `pos + size - (1, 1)` then it's the real maximum value and `inclusive_rectangle` should be used.

Reply:
got it, thanks